From: Cole Robinson <crobinso@redhat.com>
Date: Wed, 28 Apr 2010 20:07:09 +0200
Subject: [PATCH] console: Don't throw traceback if we can't read error fd.

This is racy and best effort, so don't log errors if we fail.

Origin: Upstream, http://hg.fedorahosted.org/hg/virt-manager/rev/c206b12a8c7a
---
 src/virtManager/console.py |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/src/virtManager/console.py b/src/virtManager/console.py
index cbefcb3..2333bc4 100644
--- a/src/virtManager/console.py
+++ b/src/virtManager/console.py
@@ -573,7 +573,11 @@ class vmmConsolePages(gobject.GObject):
         errfd = self.vncTunnel[1]
         errout = ""
         while True:
-            new = errfd.recv(1024)
+            try:
+                new = errfd.recv(1024)
+            except:
+                break
+
             if not new:
                 break
 
-- 
